include/buffer_tree.h

```diff
@@ -36,37 +38,45 @@ class BufferTree {
   // branching factor
   uint32_t B;
 
+  // number of nodes in the graph
+  Node N;
+
   // metadata control block(s)
-  std::vector<BufferControlBlock> buffers;
+  // level 1 blocks take indices 0->(B-1). So on and so forth from there
+  std::vector<BufferControlBlock*> buffers;
+
+  // buffers which we will use when performing flushes
+  char **flush_buffers;
```

Comment:
are these extra buffers we allocate or just what we do when we read the data into memory?

Reply:
These are page_size buffers which we use when flushing a buffer. We need these to have efficient IO

Reply:
There are a few other buffers which we should include in the buffer_tree structure such as those for when we read the data into memory for flushing or getting data.

Reply:
Can you explain why we need these for efficient IO?

Reply:
A 'flush' in this context is moving data from some node in the buffer tree to its children. These buffers allow us to have an in RAM buffer for each child so that we can scan through the updates in the parent and write them out in an efficient fashion.